modules/ingest-common/src/test/java/org/opensearch/ingest/common/URLDecodeProcessorTests.java

```diff
@@ -32,13 +32,14 @@
 
 package org.opensearch.ingest.common;
 
-import java.io.UnsupportedEncodingException;
 import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
 
 public class URLDecodeProcessorTests extends AbstractStringProcessorTestCase<String> {
     @Override
     protected String modifyInput(String input) {
-        return "Hello%20G%C3%BCnter" + input;
+        return "Hello%20G%C3%BCnter" + urlEncode(input);
```

Comment:
@andrross this is very interesting, why the test needs to manually do `urlEncode`? Shouldn't the input there be properly encoded before?

Reply:
I think the input is a random string and the test works as long as it doesn't have a '%' in it.

Reply:
Sure, the random string could be equally encoded before feeding it to `URLDecodeProcessorTests`, the part which concerns - the test itself does manipulation in order to pass, but apparently that not how the things are working.

Reply:
The test takes a random string and prefixes it with some URL-encoded content, and verifies that the result is the URL-decoded version of what was input into it. `AbstractStringProcessorTestCase` doesn't know anything about URL encoding. The previous version of this test assumed that the random part of the string was URL decode-able, which is not a valid assumption for all random strings. The easiest way to make sure the random string is decode-able is to just URL-encode it to start with. I agree it looks weird. Honestly I'm not sure the random string part is important. I could change it to just do the following:

```java
@Override
protected String modifyInput(String input) {
    return "Hello%20G%C3%BCnter";
}

@Override
protected String expectedResult(String input) {
    return "Hello Günter";
}
```

What do you think? There is in fact precedent for this approach: 
https://github.com/opensearch-project/OpenSearch/blob/9168f1fb43015cf7f33eab30f1300ec85f4d4305/modules/ingest-common/src/test/java/org/opensearch/ingest/common/GsubProcessorTests.java#L44-L52

Reply:
thanks @andrross, I had something like this in mind - generate random input in  `urlEncode` form:

```
@Override
    protected AbstractStringProcessor<String> newProcessor(String field, boolean ignoreMissing, String targetField) {
        return new URLDecodeProcessor(urlEncode(randomAlphaOfLength(10)), null, field, ignoreMissing, targetField);
    }
```

> AbstractStringProcessorTestCase doesn't know anything about URL encoding.

That's I think is what should be preserved, but using `urlEncode` breaks that

Reply:
That's not the randomness that is causing the problem though. The problem is [here](https://github.com/opensearch-project/OpenSearch/blob/37ac3788a3937ed415039edb2671da08e6450ed5/modules/ingest-common/src/test/java/org/opensearch/ingest/common/AbstractStringProcessorTestCase.java#L169), which I believe is why that `modifyInput` method was added to the base test class to provide a hook to modify what was generated.

Reply:
Oh I see @andrross , apologies for confusion, I think you change makes sense: the `modifyInput` is use to ingest the field and `URLDecodeProcessor` assumes that the field is URL encoded.